test/unit/spec/narrative_core/jobCommChannel-spec.js

```diff
@@ -128,32 +85,21 @@ define(['jobCommChannel', 'base/js/namespace', 'common/runtime'], (
                     },
                 },
             });
-            comm.initCommChannel()
-                .then(() => {
-                    expect(comm.comm).not.toBeNull();
-                    done();
-                })
-                .catch((err) => {
-                    fail(err);
-                });
+            return comm.initCommChannel().then(() => {
+                expect(comm.comm).not.toBeNull();
+            });
         });
 
-        it('Should fail to initialize with a failed reply from the JobManager startup', (done) => {
+        it('Should fail to initialize with a failed reply from the JobManager startup', async () => {
             const comm = new JobCommChannel();
             Jupyter.notebook = makeMockNotebook(null, null, {
                 name: 'Failed to start',
                 evalue: 'Some error',
                 error: 'Yes. Very yes.',
             });
-            comm.initCommChannel()
-                .then(() => {
-                    console.error('Should not have succeeded.');
-                    fail();
-                })
-                .catch((err) => {
-                    expect(err).toEqual(new Error('Failed to start:Some error'));
-                    done();
-                });
+            await expectAsync(comm.initCommChannel()).toBeRejectedWith(
+                new Error('Failed to start:Some error')
```

Comment:
we should probably edit `jobCommChannel.js` to add in a space between the error name and text. It's line 487 in the source file:
```
                                    reject(new Error(reply.content.name + ': ' + reply.content.evalue));